File: GFG/Print Diagonally/Print Diagonally.java
class Solution{
    static ArrayList<Integer> downwardDigonal(int N, int A[][])
    {
        // code here 
        ArrayList<Integer> ans=new ArrayList<>();
        int row=0;
        int col=0;
        while(col<N){
            int r=row;
            int c=col;
            while(r<N && c>=0){
                ans.add(A[r][c]);
                r++;
                c--;
            }
            col++;
        }
        col=N-1;
        row=1;
        while(row<N){
            int r=row;
            int c=col;
            while(r<N && c>=0){
                ans.add(A[r][c]);
                r++;
                c--;
            }
            row++;
        }
        return ans;
    }
}

File: GFG/Print Diagonally/Print Diagonally.cpp
class Solution{
	
	public:
	vector<int> downwardDigonal(int N, vector<vector<int>> A)
	{
		// Your code goes here
		vector<int>v;
		for(int i=0;i<N;i++){
		    for(int j=0;j<N&&j<=i;j++){
		          v.push_back(A[j][i-j]);
		    }
		}
		int s=1;
		for(int i=N;i<=2*(N-1);i++){
		    for(int j=s;j<N;j++){
		          v.push_back(A[j][i-j]);
		    }
		    s++;
		}
		return v;

	}

};
